numpy/core/multiarray.py

```diff
@@ -141,9 +141,9 @@ def empty_like(prototype, dtype=None, order=None, subok=None, shape=None):
 
 
 @array_function_from_c_func_and_dispatcher(_multiarray_umath.concatenate)
-def concatenate(arrays, axis=None, out=None):
+def concatenate(arrays, axis=None, out=None, dtype=None, casting=None):
```

Comment:
did you mean to write casting="same_kind" here ?

Reply:
No, since this is only used by the `__array_function__` mechanism, these are always just `None` to not duplicate the default value.